code/datums/components/surgery_initiator.dm

```diff
@@ -310,3 +348,50 @@
 
 /datum/component/surgery_initiator/robo/sharp
 	valid_starting_types = SURGERY_INITIATOR_ORGANIC | SURGERY_INITIATOR_ROBOTIC
+
+/datum/component/surgery_initiator/cloth
+	can_cancel = FALSE
+	surgery_start_sound = "rustle"
+
+/datum/component/surgery_initiator/cloth/show_starting_message(mob/user, mob/living/target, datum/surgery/procedure)
+	user.visible_message(
+		"<span class='notice'>[user] drapes [parent] over [target]'s [parse_zone(user.zone_selected)] to prepare for surgery.</span>",
+		"<span class='notice'>You drape [parent] over [target]'s [parse_zone(user.zone_selected)] to prepare for \an [procedure.name].</span>",
+	)
+
+/datum/component/surgery_initiator/cloth/try_choose_surgery(mob/user, mob/living/target, datum/surgery/surgery)
+	var/datum/surgery/new_procedure = ..()
+	if(!istype(new_procedure))
+		return
+
+	new_procedure.started_with_drapes = TRUE
+
+/datum/component/surgery_initiator/cloth/on_surgery_selection(mob/user, mob/living/target, datum/surgery/target_surgery)
+	user.visible_message(
+		"<span class='notice'>[user] starts to apply [parent] onto [target].</span>",
+		"<span class='notice'>You start to apply [parent] onto [target].</span>",
+	)
+
+	if(!isnull(surgery_start_sound))
+		playsound(src, surgery_start_sound, 50, TRUE)
+
+	playsound(src, surgery_start_sound)
+	if(!do_after_once(user, 3 SECONDS, TRUE, target))
+		user.visible_message(
+			"<span class='warning'>[user] stops applying [parent] onto [target].</span>",
+			"<span class='warning'>You stop applying [parent] onto [target].</span>"
+		)
+		return
+
+
+	if(!isnull(surgery_start_sound))
+		playsound(src, surgery_start_sound, 50, TRUE)
+
+	return TRUE
+
+/datum/component/surgery_initiator/cloth/on_blood_splash(datum/surgery, mob/user, mob/target, zone, obj/item/tool)
+	if(prob(90 * germ_prevention_quality))
+		target.visible_message("<span class='notice'>Blood splashes onto the dressing.</span>")
+		var/obj/item/I = parent  // safety: this component can only go onto an item
+		I.add_mob_blood(target)
+		return COMPONENT_BLOOD_SPLASH_HANDLED
```

Comment:
Doesn't this mean that the worse options are better at blocking blood splashes?

Reply:
...would it? `germ_prevention_quality` is fractional, so it'd be less likely to "handle" the splash.